addOns/pscanrulesAlpha/src/main/java/org/zaproxy/zap/extension/pscanrulesAlpha/JSFunctionPassiveScanner.java

```diff
@@ -0,0 +1,170 @@
+import net.htmlparser.jericho.HTMLElementName;
+import net.htmlparser.jericho.Source;
+import org.apache.log4j.Logger;
+import org.parosproxy.paros.Constant;
+import org.parosproxy.paros.core.scanner.Alert;
+import org.parosproxy.paros.network.HttpMessage;
+import org.zaproxy.zap.extension.pscan.PassiveScanThread;
+import org.zaproxy.zap.extension.pscan.PluginPassiveScanner;
+
+/** Passive Scan Rule for Dangerous JS Functions https://github.com/zaproxy/zaproxy/issues/5673 */
+public class JSFunctionPassiveScanner extends PluginPassiveScanner {
+
+    /** Prefix for internationalized messages used by this rule */
+    private static final String MESSAGE_PREFIX = "pscanalpha.jsfunction.";
+
+    public static final String FUNC_LIST_DIR = "xml";
+    public static final String FUNC_LIST_FILE = "js-function-list.txt";
+    private static final Logger LOGGER = Logger.getLogger(JSFunctionPassiveScanner.class);
+    private static final int PLUGIN_ID = 10110;
+
+    private static List<Pattern> patterns = null;
+    private PassiveScanThread parent = null;
+
+    @Override
+    public void scanHttpRequestSend(HttpMessage msg, int id) {
+        // This rule only scans responses received
+    }
+
+    @Override
+    public void scanHttpResponseReceive(HttpMessage msg, int id, Source source) {
+        if (patterns == null) {
+            patterns = getPatterns();
+        }
+        if (msg.getResponseBody().length() > 0 && msg.getResponseHeader().isText()) {
+            StringBuilder evidence = new StringBuilder();
+            // Check the scripts
+            Element el;
+            int offset = 0;
+            while ((el = source.getNextElement(offset, HTMLElementName.SCRIPT)) != null) {
+                String elStr = el.toString();
+                for (Pattern pattern : patterns) {
+                    if (pattern.matcher(elStr).find()) {
+                        evidence.append(elStr);
+                        evidence.append("\n");
+                        break; // Only need to record this script once
+                    }
+                }
+                offset = el.getEnd();
+            }
+            if (evidence.length() > 0) {
+                this.raiseAlert(msg, id, evidence.toString());
+            }
+        }
+    }
+
+    private void raiseAlert(HttpMessage msg, int id, String evidence) {
+        Alert alert =
+                new Alert(getPluginId(), Alert.RISK_MEDIUM, Alert.CONFIDENCE_MEDIUM, getName());
+        alert.setDetail(
+                this.getDescription(),
+                msg.getRequestHeader().getURI().toString(),
+                "", // Param, not relevant for this vulnerability
+                "", // Attack, not relevant for passive vulnerabilities
+                "", // Other info not specified in message bundle
+                this.getSolution(),
+                this.getReference(),
+                evidence,
+                0, // CWE Id - return 0 if no relevant one
```

Comment:
```suggestion
                749, // CWE-749: Exposed Dangerous Method or Function
```

Or maybe: CWE-676: Use of Potentially Dangerous Function ?